### ISSUE TITLE
Texture pixellisation at high tangential angle of view when using Custom Shaders

### ISSUE THREAD

**Issue by xtassin** [CesiumGS/cesium#10850]
I am not sure if this is a renderer or an converter issue.

I am exporting GLTF 2.0 models from 3DS max using the OpenCollada plugin and converting from DAE to GLB with COLLADA2GLTF (2.1.5)

When loaded in viewer (Cesium particularly and, to a certain measure, gltf-viewer.donmccurdy.com) texture appear pixelated when viewed from an incident angle. It looks like no anisotropic filtering is being applied. (see screenshots)

![texure-gltf2](https://user-images.githubusercontent.com/296572/195007537-303123dc-aa3e-4296-a771-fe2d357314ac.jpg)

**Comment by ggetz** [comment]
Thanks @xtassin! It may help if you could provide the model, either here in GitHub or to support@cesium.com.

@lilleyse Would you be able to take a look at whether this is a viewer issue or a converter issue?

**Comment by lilleyse** [comment]
Do the glTF 1 and glTF 2 models have different samplers? Maybe this is mimap related?

Both the old and new model implementations set `TEXTURE_MAX_ANISOTROPY_EXT` to 1.

**Comment by xtassin** [comment]
Ok, I had written a very detailed experimentation process that I will leave at the bottom, for reference, and just because it costed me quite some time to write ;)

But the conclusion is that the latest COLLADA2GLTF exporter (v 2.1.5) is setting the sampler minification filter to 9986 (NearestMipmapLinear) instead of 9729 (Linear) in the older version.

I believe we should have control on that from the arguments passed to the exporter.

--------------------------------------------------------------

Some more experimenting points at an converter problem.
Here are three examples (see zip file attached): 

![image](https://user-images.githubusercontent.com/296572/195542289-5b4d2899-b2bb-4f78-a970-a3ca6d90dd99.png)

All using the same external JPG texture
All are converted from the same DAE (3DS max export - OpenCollada plugin - with normals, no tangents)

**cube-1.0-OldExporter.gltf** 
- produced with an older version of COLLADA2GLTF. I can not really tell which but I am pretty sure it is < 2.0. Calling the exe with -v option returns collada2gltf@6779c094d694e951beeb1012d03d4faa1edb0ddf
- Command line (old format): collada2gltf.exe -e -r -f

**cube-1.0-LatestExporter.gltf**
- produced with latest COLLADA2GLTF 2.1.5
- command line: COLLADA2GLTF-bin.exe -t -v 1.0 -m
- Note that exporting to GLTF 1.0 without the -m option will lead to a shader exception in Cesium:
```
RuntimeError: Fragment shader failed to compile.  Compile log: ERROR: 0:18: 'texture2D' : no matching overloaded function found
ERROR: 0:18: '=' : dimension mismatch
ERROR: 0:18: '=' : cannot convert from 'const mediump float' to 'highp 4-component vector of float'
```

**cube-2.0-LatestExporter.gltf**
- produced with latest COLLADA2GLTF 2.1.5
- command line: COLLADA2GLTF-bin.exe -b -t

[gltf.zip](https://github.com/CesiumGS/cesium/files/9772446/gltf.zip)

**Comment by xtassin** [comment]
Closing this issue as this is mostly linked to the old COLLADA2GLTF converter which appears not to be supported anymore.
- Using other/newer GLTF exporter together with properly configured custom shaders prevents the texture filtering issue.
- The GLSL exception I reported above is clearly due to old shaders (WebGL1) being generated by the old converter while running a WebGL2 context with the latest Cesium release.
